MapboxNavigation/RatingControl.swift

```diff
@@ -0,0 +1,140 @@
+import UIKit
+import CoreGraphics
+
+enum RatingControlError: String, Error {
+    case tappedButtonNotInArray = "The button sending the tapped event is not in the internal button array."
+}
+
+typealias RatingClosure = (Int) -> Void //rating
+
+/*@IBDesignable*/
+class RatingControl: UIStackView {
+    
+    //MARK: Constants
+    static let defaultSize = CGSize(width: 32.0, height: 32.0)
+    private let starTemplate = UIImage(named: "star", in: .mapboxNavigation, compatibleWith: nil)
+    
+    
+    //MARK: Properties
+    private var stars = [UIButton]()
+    
+    var didChangeRating: RatingClosure?
+    
+    var rating: Int = 0 {
+        didSet {
+            updateSelectionStates()
+            didChangeRating?(rating)
+        }
+    }
+    
+    @IBInspectable var selectedColor: UIColor = #colorLiteral(red: 0.1205472574, green: 0.2422055006, blue: 0.3489340544, alpha: 1)
+    @IBInspectable var normalColor: UIColor = #colorLiteral(red: 0.8508961797, green: 0.8510394692, blue: 0.850877285, alpha: 1)
```

Comment:
It'd be nice to have this view be style-able one day. Would it make sense to inherit colors from other Style components? Or does it make more sense to pull this out into it's own component?

/cc @frederoni

Reply:
We can focus on this last btw.

Reply:
As we're slowly moving away from storyboards I'd prefer designable views but not inspectable properties. It's very easy to leave a stray UDRA (User Defined Runtime Attribute) in the storyboard which would lead to a crash since there's no compile-time check. Public class and dynamic public var would make it stylable.

Reply:
That would only be a risk if a developer customizes the color in a version of this library that uses a storyboard but later upgrades, correct? Perhaps we can treat adding color inspectables as tail work for when the migration away from storyboards is complete.

Reply:
- [ ] Do this